moveit_ros/trajectory_cache/src/utils/utils.cpp

```diff
@@ -0,0 +1,659 @@
+  out.max_step = max_step;
+  out.jump_threshold = jump_threshold;
+  out.path_constraints = moveit_msgs::msg::Constraints();
+  out.avoid_collisions = avoid_collisions;
+  out.link_name = move_group.getEndEffectorLink();
+  out.header.stamp = move_group.getNode()->now();
+
+  return out;
+}
+
+// Queries. ========================================================================================
+
+void queryAppendCenterWithTolerance(Query& query, const std::string& name, double center, double tolerance)
+{
+  query.appendRangeInclusive(name, center - tolerance / 2, center + tolerance / 2);
+}
+
+// Constraints. ====================================================================================
+
+void sortJointConstraints(std::vector<moveit_msgs::msg::JointConstraint>& joint_constraints)
+{
+  std::sort(joint_constraints.begin(), joint_constraints.end(),
+            [](const moveit_msgs::msg::JointConstraint& l, const moveit_msgs::msg::JointConstraint& r) {
+              return l.joint_name < r.joint_name;
+            });
+}
+
+void sortPositionConstraints(std::vector<moveit_msgs::msg::PositionConstraint>& position_constraints)
+{
+  std::sort(position_constraints.begin(), position_constraints.end(),
+            [](const moveit_msgs::msg::PositionConstraint& l, const moveit_msgs::msg::PositionConstraint& r) {
+              return l.link_name < r.link_name;
+            });
+}
+
+void sortOrientationConstraints(std::vector<moveit_msgs::msg::OrientationConstraint>& orientation_constraints)
+{
+  std::sort(orientation_constraints.begin(), orientation_constraints.end(),
+            [](const moveit_msgs::msg::OrientationConstraint& l, const moveit_msgs::msg::OrientationConstraint& r) {
+              return l.link_name < r.link_name;
+            });
+}
+
+moveit::core::MoveItErrorCode
+appendConstraintsAsFetchQueryWithTolerance(Query& query, std::vector<moveit_msgs::msg::Constraints> constraints,
+                                           const MoveGroupInterface& move_group, double match_tolerance,
+                                           const std::string& reference_frame_id, const std::string& prefix)
+{
+  const std::shared_ptr<tf2_ros::Buffer> tf_buffer = move_group.getTF();  // NOLINT: Deliberate lifetime extension.
+
+  // Make ignored members explicit.
+
+  bool emit_position_constraint_warning = false;
+  for (auto& constraint : constraints)
+  {
+    for (auto& position_constraint : constraint.position_constraints)
+    {
+      if (!position_constraint.constraint_region.primitives.empty())
+      {
+        emit_position_constraint_warning = true;
+        break;
+      }
+    }
+    if (emit_position_constraint_warning)
+    {
+      break;
+    }
+  }
+  if (emit_position_constraint_warning)
+  {
+    RCLCPP_WARN_STREAM(getLogger(), "Ignoring " << prefix << ".position_constraints.constraint_region: Not supported.");
+  }
+
+  bool emit_visibility_constraint_warning = false;
+  for (auto& constraint : constraints)
```

Comment:
I am only skimming this code, but is there any chance that some of these range based for loops can use `const auto&` instead of regular `auto&`?

Reply:
I made them all const except for two instances. See comment added in the commit: b495cc0356110e8dd4b5ed38893a7be8f457056d